tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -4274,3 +4559,59 @@ def from_learner(cls, learner: AbstractTabularLearner):
         predictor = cls(label=learner.label, path=learner.path)
         predictor._set_post_fit_vars(learner=learner)
         return predictor
+
+
+def _sub_fit(self, train_data, time_limit, ds_fit_kwargs, ag_fit_kwargs, ag_post_fix_kwargs, custom_val_data=None):
+    holdout_frac = ds_fit_kwargs.get("holdout_frac", None)
+    train_indices = ds_fit_kwargs.get("train_indices", None)
+    if holdout_frac is not None:
+        train_data, val_data = generate_train_test_split_combined(
+            train_data,
+            label=self.label,
+            problem_type=self.problem_type,
+            test_size=holdout_frac,
+            random_state=42,
+        )
+    elif train_indices is not None:
+        val_indices = ds_fit_kwargs.get("val_indices", None)
+        val_data = train_data.iloc[val_indices]
+        train_data = train_data.iloc[train_indices]
+    elif custom_val_data is not None:
+        train_data = train_data
+        val_data = custom_val_data
+    else:
+        raise ValueError("Unsupported validation procedure during dynamic stacking!")
+
+    set_logger_verbosity(ag_fit_kwargs["verbosity"])
+    org_learner = copy.deepcopy(self._learner)
+
+    ag_fit_kwargs["X"] = train_data
+    ag_fit_kwargs["time_limit"] = time_limit
+    ds_fit_context = ds_fit_kwargs.get("ds_fit_context")
+    clean_up_fits = ds_fit_kwargs.get("clean_up_fits")
+
+    self._learner.set_contexts(ds_fit_context)
+    self._fit(ag_fit_kwargs, ag_post_fix_kwargs)
```

Comment:
`ag_post_fix` is this a typo for `ag_post_fit`?

Reply:
always call by keyword arguments. Improves readability and reduces risk of bugs.